ledger/participant-integration-api/src/main/scala/platform/apiserver/services/ApiSubmissionService.scala

```diff
@@ -268,10 +269,14 @@ private[apiserver] final class ApiSubmissionService private[services] (
 
   private def toStatus(errorCause: ErrorCause) =
     errorCause match {
-      case e: ErrorCause.DamlLf =>
-        Status.INVALID_ARGUMENT.withDescription(e.explain)
-      case e: ErrorCause.LedgerTime =>
-        Status.ABORTED.withDescription(e.explain)
+      case cause @ ErrorCause.DamlLf(error) =>
+        error match {
+          case ContractNotFound(_) | ReplayMismatch(_) =>
```

Comment:
Why was `ReplayMismatch` absent? Is it tested?

Reply:
Samir suggested putting it here. Will add a test